src/cli/rustup_mode.rs

```diff
@@ -269,10 +272,11 @@ pub(crate) fn cli() -> Command<'static> {
             Arg::new("+toolchain")
                 .help("release channel (e.g. +stable) or custom toolchain to set override")
                 .value_parser(|s: &str| {
+                    use clap::{Error, error::ErrorKind};
                     if let Some(stripped) = s.strip_prefix('+') {
-                        ResolvableToolchainName::try_from(stripped).map_err(|e| clap::Error::raw(clap::ErrorKind::InvalidValue, e))
+                        ResolvableToolchainName::try_from(stripped).map_err(|e| Error::raw(ErrorKind::InvalidValue, e))
                     } else {
-                        Err(clap::Error::raw(clap::ErrorKind::InvalidSubcommand, format!("\"{s}\" is not a valid subcommand, so it was interpreted as a toolchain name, but it is also invalid. {TOOLCHAIN_OVERRIDE_ERROR}")))
+                        Err(Error::raw(ErrorKind::InvalidSubcommand, format!("\"{s}\" is not a valid subcommand, so it was interpreted as a toolchain name, but it is also invalid. {TOOLCHAIN_OVERRIDE_ERROR}")))
                     }
```

Comment:
Honestly, I'm surprised this works.  The intention was to support any user errors.  I'm surprised that clap errors render well when nested.  I wouldn't be surprised if we accidentally break this

Reply:
What should we do instead?

Reply:
Use any normal error type that implements `Error`

Reply:
Ok, so I don't need to use clap error types in those helpers?

Reply:
Correct.  If curious, [this](https://docs.rs/clap/latest/clap/builder/trait.TypedValueParser.html#impl-TypedValueParser-for-F) is the relevant trait implementation you need to match